src/hooks/index.ts

```diff
@@ -971,3 +940,65 @@ export const useBalances = (wallet: YoroiWallet): YoroiAmounts => {
 
   return Utxos.toAmounts(utxos, primaryTokenId)
 }
+
+export const AUTH_METHOD_KEY = '/appSettings/authMethod'
+export const useAuthMethod = (storage: Storage, options?: UseQueryOptions<AuthMethodState, Error>) => {
+  const query = useQuery({
+    suspense: true,
+    queryKey: ['authMethod'],
+    queryFn: async () => {
+      const authMethod: AuthMethod | '' = await Promise.resolve(AUTH_METHOD_KEY)
+        .then(storage.getItem)
+        .then((storedAuthMethod) => (!isEmptyString(storedAuthMethod) ? storedAuthMethod : '""'))
+        .then(JSON.parse)
+
+      if (authMethod === 'pin')
+        return {
+          PIN: true,
+          OS: false,
+          None: false,
+        } as const
+
+      if (authMethod === 'os')
+        return {
+          PIN: false,
+          OS: true,
+          None: false,
+        } as const
+
+      return {
+        PIN: false,
+        OS: false,
+        None: true,
+      } as const
+    },
+    ...options,
+  })
+
+  // refetch on settings
+  return {
```

Comment:
suspense query, return the value directly

Reply:
⬆️